packages/suite/src/hooks/wallet/useCoinmarket.ts

```diff
@@ -18,49 +20,58 @@ export const useInvityAPI = () => {
         sellInfo,
         invityAPIUrl,
         exchangeCoinInfo,
+        isLoading,
+        lastLoadedTimestamp,
     } = useSelector(state => ({
         selectedAccount: state.wallet.selectedAccount,
         buyInfo: state.wallet.coinmarket.buy.buyInfo,
         exchangeInfo: state.wallet.coinmarket.exchange.exchangeInfo,
         sellInfo: state.wallet.coinmarket.sell.sellInfo,
         invityAPIUrl: state.suite.settings.debug.invityAPIUrl,
         exchangeCoinInfo: state.wallet.coinmarket.exchange.exchangeCoinInfo,
+        isLoading: state.wallet.coinmarket.isLoading,
+        lastLoadedTimestamp: state.wallet.coinmarket.lastLoadedTimestamp,
     }));
 
-    const { saveBuyInfo, saveExchangeInfo, saveExchangeCoinInfo, saveSellInfo } = useActions({
+    const {
+        saveBuyInfo,
+        saveExchangeInfo,
+        saveExchangeCoinInfo,
+        saveSellInfo,
+        setIsLoading,
+    } = useActions({
         saveBuyInfo: coinmarketBuyActions.saveBuyInfo,
         saveExchangeInfo: coinmarketExchangeActions.saveExchangeInfo,
         saveExchangeCoinInfo: coinmarketExchangeActions.saveExchangeCoinInfo,
         saveSellInfo: coinmarketSellActions.saveSellInfo,
+        setIsLoading: coinmarketCommonActions.setLoading,
     });
 
     if (selectedAccount.status === 'loaded') {
         if (invityAPIUrl) {
             invityAPI.setInvityAPIServer(invityAPIUrl);
         }
 
-        invityAPI.createInvityAPIKey(selectedAccount.account?.descriptor);
-
-        if (!buyInfo?.buyInfo) {
-            coinmarketBuyActions.loadBuyInfo().then(buyInfo => {
-                saveBuyInfo(buyInfo);
-            });
-        }
-
-        if (!exchangeInfo) {
-            coinmarketExchangeActions
-                .loadExchangeInfo()
-                .then(([exchangeInfo, exchangeCoinInfo]) => {
+        invityAPI.createInvityAPIKey(selectedAccount.account.descriptor);
+
+        if (!isLoading && lastLoadedTimestamp + InvityAPIReloadDataAfterMs < Date.now()) {
+            setIsLoading(true);
+            Promise.all([
+                coinmarketBuyActions.loadBuyInfo(),
+                coinmarketExchangeActions.loadExchangeInfo(),
+                coinmarketSellActions.loadSellInfo(),
+            ])
+                .then(([buyInfoResult, [exchangeInfo, exchangeCoinInfo], sellInfoResult]) => {
+                    setIsLoading(false, Date.now());
+                    saveBuyInfo(buyInfoResult);
                     saveExchangeInfo(exchangeInfo);
                     saveExchangeCoinInfo(exchangeCoinInfo);
+                    saveSellInfo(sellInfoResult);
+                })
+                .catch(() => {
+                    setIsLoading(false);
```

Comment:
i'm wondering on the impact of this.
example: if loading of `exchangeInfo` fails then you will not have ANY other info (sell, buy..) it this ok?

another thing is architecture of this code, should it be in hook? (i know it was like this before)
but maybe it should be moved to some middleware which is independent from the react render process (and from mount/unmount events) ?

Reply:
Agreed, will fix.
Any chance to upgrade TS lib es2019 -> es2020?
So we can use `Promise.allSettled` for easier handling of fulfilled or rejected promises?

Reply:
good question, maybe with the incoming new pipeline?
but it doesn't look like it happen anytime soon... maybe leave some TODO note there